### ISSUE TITLE
JVM 11+ support

### ISSUE THREAD

**Issue by ov7a** [Anamorphosee/stacktrace-decoroutinator#7]
Does this library work with JVM 11, 17, etc?
For example, with corretto-17.0.3 and stacktrace-decoroutinator-jvm:2.2.1 I get 
```
Exception in thread "main" java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.findLoadedClass(java.lang.String) accessible: module java.base does not "opens java.lang" to unnamed module @1c20c684
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at dev.reformator.stacktracedecoroutinator.utils.Utils_jvmKt.getClassIfLoaded(utils-jvm.kt:11)
	at dev.reformator.stacktracedecoroutinator.runtime.DecoroutinatorRuntime.getState(runtime-jvm.kt:42)
	at dev.reformator.stacktracedecoroutinator.runtime.DecoroutinatorRuntime.load(runtime-jvm.kt:52)
	at dev.reformator.stacktracedecoroutinator.runtime.DecoroutinatorRuntime.load$default(runtime-jvm.kt:51)
	at MainKt.main(Main.kt:14)
	at MainKt.main(Main.kt)
```
when I launch the example from README. With 1.8 it works fine.

**Comment by Anamorphosee** [comment]
please check version 2.3.0. It have to fix this issue.

**Comment by ov7a** [comment]
The example works with 2.3.0, yeah, thanks. 

With bigger project I get a strange exception when I just add a dependency and run a junit test. 
```
Caused by: java.lang.IllegalStateException: Decoroutinator registry is not set. Didn't you miss to call 'DecoroutinatorRuntime.load()'?
	at dev.reformator.stacktracedecoroutinator.common.Registry_commonKt.decoroutinatorRuntimeNotLoaded(registry-common.kt:36)
	at dev.reformator.stacktracedecoroutinator.common.Registry_commonKt.access$decoroutinatorRuntimeNotLoaded(registry-common.kt:1)
	at dev.reformator.stacktracedecoroutinator.common.Registry_commonKt$decoroutinatorRegistry$1.getEnabled(registry-common.kt:29)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(continuation-stdlib.kt:17)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	... 90 more
```
I think a common use case is when decoroutinator is enabled/disabled in test/prod environment via debug switch, and just having a dependency shouldn't impact project behaviour. Unfortunately, I cannot reproduce this in sample project, maybe I'll dig into this later.

**Comment by Anamorphosee** [comment]
the last exception throws when you doesn't call method 'DecoroutinatorRuntime.load()'.
For unit tests you should add that call in `@BeforeTest` static method. Example is at https://github.com/Anamorphosee/stacktrace-decoroutinator/blob/master/stacktrace-decoroutinator-jvm/src/test/kotlin/runtime-test-jvm.kt#L26.

**Comment by ov7a** [comment]
I get it, but what should a use do if he wants to switch Decoroutinator on and off via flag (config setting, env variable, etc)? For example, to enable it in tests and in dev environments, but disable it in prod?

**Comment by Anamorphosee** [comment]
Right now you can set the system property `dev.reformator.stacktracedecoroutinator.enabled` to `false` before calling `DecoroutinatorRuntime.load()` to disable SD. But the call is still mandatory.

This property is read once, so you cannot switch SD on and off dynamically during the execution of your app.

**Comment by ov7a** [comment]
Its nice to have this in README. 

But even with systemProperty set to false, decoroutinator still impacts the project. Behavior with and without dependency differs. For example, mockito complains and can't mock some objects (didn't dig that yet).

**Comment by Anamorphosee** [comment]
I'm agree with you that just existence of the dependency shouldn't change the app behaviour. It will be fixed in #8.

**Comment by Anamorphosee** [comment]
Thank you for the feedback!
